Sources/ProcedureKit/ProcedureQueue.swift

```diff
@@ -137,129 +184,273 @@ open class ProcedureQueue: OperationQueue {
+            if let previous = manager.add(procedure: procedure, category: category) {
+                previousMutuallyExclusiveOperations.insert(previous)
             }
-
-            // Add the evaluator
-            addOperation(evaluator)
         }
 
-        // Indicate to the operation that it is to be enqueued
-        procedure.willEnqueue(on: self)
+        // Create the condition evaluator
+        let evaluator = procedure.evaluateConditions()
 
-        delegate?.procedureQueue(self, willAddOperation: procedure)
-    }
-    // swiftlint:enable cyclomatic_complexity
-    // swiftlint:enable function_body_length
+        // Get the condition dependencies
+        let indirectDependencies = procedure.indirectDependencies
 
+        // If there are dependencies
+        if indirectDependencies.count > 0 {
 
-    /**
-     Adds the operations to the queue.
+            // Filter out the indirect dependencies which have already been added to the queue
+            let indirectDependenciesToProcess = indirectDependencies.filter { !self.operations.contains($0) }
 
-     - parameter ops: an array of `NSOperation` instances.
-     - parameter wait: a Bool flag which is ignored.
+            // Check to see if there are any which need processing
+            if indirectDependenciesToProcess.count > 0 {
+
+                // Iterate through the indirect dependencies
+                indirectDependenciesToProcess.forEach {
+
+                    // Indirect dependencies are executed after
+                    // any previous mutually exclusive operation(s)
+                    $0.add(dependencies: previousMutuallyExclusiveOperations)
+
+                    // Indirect dependencies are executed after
+                    // all direct dependencies
+                    $0.add(dependencies: procedure.directDependencies)
+
+                    // Only evaluate conditions after all indirect
+                    // dependencies have finished
+                    evaluator.addDependency($0)
+                }
+
+                // Add indirect dependencies
+                add(operations: indirectDependenciesToProcess)
+            }
+        }
+
+        return evaluator
+    }
+}
+
+public extension ProcedureQueue {
+    /**
+     Add operations to the queue as an array
+     
+     - parameters operations: a variadic array of `NSOperation` instances.
+     - parameter context: an optional parameter that is passed-through to the Will/DidAdd delegate callbacks
+     - returns: a ProcedureFuture that is signaled once the operation has been added to the ProcedureQueue
      */
-    open override func addOperations(_ ops: [Operation], waitUntilFinished wait: Bool) {
-        ops.forEach { addOperation($0) }
+    @discardableResult
+    final func add<S: Sequence>(operations: S, withContext context: Any? = nil) -> ProcedureFuture where S.Iterator.Element: Operation {
+
+        let futures = Array(operations).map {
```

Comment:
No need to create an array first if all you're doing is a map, `Sequence` has a `map` that returns an `Array`.